glue_vispy_viewers/common/vispy_widget.py

```diff
@@ -77,6 +114,20 @@ def __init__(self, parent=None):
         add_callback(self, 'visible_axes', nonpartial(self._toggle_axes))
         add_callback(self, 'perspective_view', nonpartial(self._toggle_perspective))
 
+    # TODO: how to get data here?
+    def update_axis_label(self, data):
+        # data object
+        label = []
+        tick_value = []
+        if data.coords:
+            # TODO: think about 4th dim
+            for i in range(data.ndim):
+                label.append(data.coords.axis_label(i))  # ['Vopt', 'Declination', 'Right Ascension']
+                tick_value.append(data._world_component_ids[label[i]])
+            self.xax.domain = tick_value[0]
+            self.yax.domain = tick_value[1]
+            self.zax.domain = tick_value[2]
+
```

Comment:
Actually it's an easy fix - just get rid of the ``update_axis_label`` method completely and instead put the following code:

```
        self.xax.domain = (self.options.x_min, self.options.x_max)
        self.yax.domain = (self.options.y_min, self.options.y_max)
        self.zax.domain = (self.options.z_min, self.options.z_max)
```

at the end of the ``_update_limits`` method. This method gets called whenever the limits are updated, and the above code basically just takes the limits from the options widget. Let's not worry about the axis labels for now, we should try and get this working first. Once I make the above fix, the values on the axes look right.

There are a few remaining esthetic issues:

* The ticks are a little too long - maybe half the length would be better (both for minor and major ticks)

* The labels themselves are also a little large

* The ticks and labels on the x and y axes are fine, but the ones of the z axis don't appear quite in the right place - I think they need to be rotated (around the z axis) so that they point outward from the cube.

Could you make these small fixes, and then I think we can merge this in and try and add axis labels in a separate pull request? (that is, the name of the axes)